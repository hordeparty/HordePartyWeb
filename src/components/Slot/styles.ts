import styled from 'styled-components'

export const Wrapper = styled.div`
  width: 20rem;
  height: 3rem;
  border: 5px solid #000;
  border-radius: 5px;
  display: flex;

  > div:first-child {
    border-right: 5px solid #000;
    width: 15rem;
  }
`

export const Divider = styled.div`
  width: 10rem;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-evenly;
`

export const Dot = styled.div`
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background-color: #000;
`
