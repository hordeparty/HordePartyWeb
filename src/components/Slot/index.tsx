import * as S from './styles'

const Slot = () => (
  <S.Wrapper>
    <S.Divider>
      <S.Dot />
      <S.Dot />
      <S.Dot />
      <S.Dot />
    </S.Divider>
    <S.Divider>
      <S.Dot />
      <S.Dot />
      <S.Dot />
    </S.Divider>
  </S.Wrapper>
)

export default Slot
