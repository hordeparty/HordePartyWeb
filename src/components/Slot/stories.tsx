import { Story, Meta } from '@storybook/react'
import Slot from '.'

export default {
  title: 'Slot',
  component: Slot
} as Meta

export const Default: Story = () => <Slot />
