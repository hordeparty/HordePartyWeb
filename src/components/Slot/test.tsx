import { render } from '@testing-library/react'

import Slot from '.'

describe('<Slot />', () => {
  it('should render the Slot component', () => {
    const { container } = render(<Slot />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
