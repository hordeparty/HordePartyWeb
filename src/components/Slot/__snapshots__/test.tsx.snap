// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slot /> should render the Slot component 1`] = `
.c0 {
  width: 20rem;
  height: 3rem;
  border: 5px solid #000;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > div:first-child {
  border-right: 5px solid #000;
  width: 15rem;
}

.c1 {
  width: 10rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c2 {
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background-color: #000;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
    <div
      class="c2"
    />
    <div
      class="c2"
    />
    <div
      class="c2"
    />
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
    <div
      class="c2"
    />
    <div
      class="c2"
    />
  </div>
</div>
`
